### PR TITLE
docs(guides): replace all rules and synonyms

### PR DESCRIPTION
**Pull request by aallam**
🎟 JIRA Ticket: APIC-518

### Changes included:

Add guide to replace all rules and synonyms using `saveRules` and `saveSynonyms` methods respectively.

[Preview](https://deploy-preview-1668--api-clients-automation.netlify.app/docs/clients/guides/replace-all-rules-synonyms)

### CODE REVIEW

File: website/docs/clients/guides/replace-all-rules-synonyms.mdx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+// TODO
+
+</TabItem>
+<TabItem value="java">
+
+java
+
+// TODO
+
+</TabItem>
```
Comment (Fluf22) [inline website/docs/clients/guides/replace-all-rules-synonyms.mdx:39]: Should we wait for those examples, or do you want to go with them as-is?

File: website/docs/clients/guides/replace-all-rules-synonyms.mdx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+In Algolia, you can replace all existing rules and synonyms by leveraging the appropriate methods. This guide will show you how to use `saveRules` and `saveSynonyms` methods to achieve this.
+
+## Replace All Rules
+
+To replace all existing rules in Algolia, you can use the `saveRules` method with the `clearExistingRules` parameter set to `true`.
+
+<TabsLanguage>
+<TabItem value="javascript">
+
```
Comment (Fluf22) [inline website/docs/clients/guides/replace-all-rules-synonyms.mdx:16]: Is there no indentation possible? 😬

File: website/docs/clients/guides/replace-all-rules-synonyms.mdx [hunk 3]
```diff
@@ -0,0 +1,80 @@
+---
+title: Replace All Rules and Synonyms
+---
+
+import { TabsLanguage } from '../../../src/components/TabsLanguage';
```
Comment (Fluf22) [inline website/docs/clients/guides/replace-all-rules-synonyms.mdx:5]: In the future, we could add some base directory, so we can get rid of those `../` 😁

**Review by Fluf22 — COMMENTED** [review]
Since @shortcuts is in PTO, I took the responsability to add a lot of comments 😇 😚 😂

File: website/docs/clients/guides/replace-all-rules-synonyms.mdx (hunk 3 above)
Comment (aallam) [reply website/docs/clients/guides/replace-all-rules-synonyms.mdx:5]: Yeah, maybe we can improve this indeed.

File: website/docs/clients/guides/replace-all-rules-synonyms.mdx (hunk 2 above)
Comment (aallam) [reply website/docs/clients/guides/replace-all-rules-synonyms.mdx:16]: I've just followed the same style as the others guides, so I am not sure!

File: website/docs/clients/guides/replace-all-rules-synonyms.mdx (hunk 1 above)
Comment (aallam) [reply website/docs/clients/guides/replace-all-rules-synonyms.mdx:39]: We will add them later, we did the same for the other guides :)

File: website/docs/clients/guides/replace-all-rules-synonyms.mdx (hunk 2 above)
Comment (Fluf22) [reply website/docs/clients/guides/replace-all-rules-synonyms.mdx:16]: Well, let's leave them as-is, then

**Review by Fluf22 — APPROVED** [review]

**Pull request merged**